official/recommendation/ncf_main.py

```diff
@@ -201,100 +210,135 @@ def main(_):
+
+    # Benchmark the evaluation results
+    benchmark_logger.log_evaluation_result(eval_results)
+    # Log the HR and NDCG results.
+    hr = eval_results[_HR_KEY]
+    ndcg = eval_results[_NDCG_KEY]
     tf.logging.info(
-        "Iteration {} [{:.1f} s]: HR = {:.4f}, NDCG = {:.4f}, [{:.1f} s]"
-        .format(cycle_index, train_cycle_end - train_cycle_begin, hr, ndcg,
-                eval_cycle_end - eval_cycle_begin))
-
-  # Remove temporary files
-  os.remove(constants.TRAIN_DATA)
-  os.remove(constants.TEST_DATA)
+        "Iteration {}: HR = {:.4f}, NDCG = {:.4f}".format(
+            cycle_index + 1, hr, ndcg))
+
+    # If some evaluation threshold is met
+    if model_helpers.past_stop_threshold(FLAGS.hr_threshold, hr):
+      break
+
+  # Clear the session explicitly to avoid session delete error
+  tf.keras.backend.clear_session()
+
+
+def define_ncf_flags():
+  """Add flags for running ncf_main."""
+  # Add common flags
+  flags_core.define_base(export_dir=False)
+  flags_core.define_performance(
+      num_parallel_calls=False,
+      inter_op=False,
+      intra_op=False,
+      synthetic_data=False,
+      max_train_steps=False,
+      dtype=False
+  )
+  flags_core.define_benchmark()
+
+  flags.adopt_module_key_flags(flags_core)
+
+  flags_core.set_defaults(
+      model_dir="/tmp/ncf/",
+      data_dir="/tmp/movielens-data/",
+      train_epochs=2,
+      batch_size=256,
+      hooks="ProfilerHook")
+
+  # Add ncf-specific flags
+  flags.DEFINE_enum(
+      name="dataset", short_name="ds", default="ml-1m",
+      enum_values=["ml-1m", "ml-20m"], case_sensitive=False,
+      help=flags_core.help_wrap(
+          "Dataset to be trained and evaluated. Two datasets are "
+          "available for now: ml-1m and ml-20m."))
+
+  flags.DEFINE_integer(
+      name="num_factors", short_name="nf", default=8,
+      help=flags_core.help_wrap("The Embedding size of MF model."))
+
+  flags.DEFINE_list(
+      name="layers", short_name="ly", default=[64, 32, 16, 8],
+      help=flags_core.help_wrap("The size of hidden layers for MLP."))
+
+  flags.DEFINE_float(
+      name="mf_regularization", short_name="mr", default=0.0,
+      help=flags_core.help_wrap("The Regularization for MF embeddings."))
+
+  flags.DEFINE_integer(
+      name="num_neg", short_name="nn", default=4,
+      help=flags_core.help_wrap(
+          "The Number of negative instances to pair with a positive instance."))
+
+  flags.DEFINE_float(
+      name="learning_rate", short_name="lr", default=0.001,
+      help=flags_core.help_wrap("The learning rate."))
+
+  flags.DEFINE_float(
+      name="hr_threshold", short_name="ht", default=None,
+      help=flags_core.help_wrap(
+          "If passed, training will stop when the evaluation metric HR is "
+          "greater than or equal to hr_threshold. For dataset ml-1m, the "
+          "desired hr_threshold is 0.68; For dataset ml-20m, the threshold can "
+          "be set as 0.95."))
```

Comment:
Are these set auto-magically? They should be, and that should be clear from the help text as well.